src/test/java/nl/ordina/jobcrawler/service/SkillServiceTest.java

```diff
@@ -0,0 +1,298 @@
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+
+class SkillServiceTest {
+
+    private SkillRepository skillRepositoryMock;
+    private SkillService skillService;
+
+    private Vacancy vacancy;
+    private List<Vacancy> vacanciesDB;  // fake "database" for vacancy
+    private String skillName;
+    private Skill skill;
+    private List<Skill> skillDB;    // fake "database" for skill
+
+    @BeforeEach
+    void init() {
+        this.skillRepositoryMock = Mockito.mock(SkillRepository.class);
+        this.skillService = new SkillService(skillRepositoryMock);
+
+        vacancy = Vacancy.builder() // making a vacancy
+                .id(UUID.randomUUID())
+                .vacancyURL("example.com")
+                .title("job title Example 1")
+                .broker("broker Example")
+                .vacancyNumber("1")
+                .hours("30")
+                .location("location example")
+                .postingDate("14 April 2020")
+                .about("this is a description of the example job")
+                .skills(new HashSet<Skill>())
+                .build();
+
+        skillName = "example";
+        skill = Skill.builder()
+                .id(UUID.randomUUID())
+                .name(skillName)
+                .vacancies(new HashSet<Vacancy>())
+                .build();
+
+        vacancy.addSkill(skill);    // making the link between the vacancy and the skill
+
+        skillDB = new ArrayList<Skill>();
+        vacanciesDB = new ArrayList<Vacancy>();
+
+        vacanciesDB.add(vacancy);   // putting the vacancy in the fake vacancy "database"
+        skillDB.add(skill); // putting the vacancy in the fake skill "database"
+    }
+
+    @Test
+    void getAllSkills() {
+        /* uses from repository methods: findAll */
+
+        Mockito.when(skillRepositoryMock.findAll()).thenReturn(skillDB);
+        assertThat(skillService.getAllSkills()).isEqualTo(skillDB);
+    }
+
+
+    @Test
+    void getExistingSkillByName() {
+        /* uses from repository methods: findByName */
+        Mockito.when(skillRepositoryMock.findByName(skillName)).thenReturn(Optional.ofNullable(skill));
+        assertThat(skillService.getSkillByName(skill).get()).isEqualTo(skill);
+    }
+
+    @Test
+    void getNonExistingSkillByName() {
+        /* uses from repository methods: findByName */
+        Mockito.when(skillRepositoryMock.findByName(skillName)).thenReturn(Optional.empty());
+        assertFalse(skillService.getSkillByName(skill).isPresent());
+    }
+
+    @Test
+    void getVacanciesBySkill() {
+        /* uses from repository methods: findByName */
+        Mockito.when(skillRepositoryMock.findByName(skillName)).thenReturn(Optional.ofNullable(skill));
+        Set<Vacancy> result = skillService.getVacanciesBySkill(skillName);
+        assertTrue(result.contains(vacancy)
+                && result.size() == 1);
```

Comment:
These are two asserts packed into one. Please unpack.

Reply:
done in https://github.com/mostafa-221/jobcrawler-backend/pull/17/commits/a97eead2dd4fc7a870d7ffd3c65686db7efbc7f7. 
During the TDD training we got the tip to only have one assert statement per test. I tried to adhere to this rule before but got these two asserts that are packed into one. what is the standard for this? readable code with multiple asserts or one assert with less readable code? or maybe something else entirely?